egs2/TEMPLATE/asr1/pyscripts/utils/extract_xvectors.py

```diff
@@ -102,13 +131,26 @@ def rawnet_extract_embd(self, audio, n_samples=48000, n_segments=10):
             output = self.model(audios)
         return output.mean(0).detach().cpu().numpy()
 
+    def _espnet_extract_embd(self, audio):
+        if len(audio.shape) > 1:
+            raise ValueError(
+                "Not support multi-channel input for ESPnet pre-trained model"
+                f"Input data has a shape of {audio.shape}."
+            )
+        audio = torch.from_numpy(audio.astype(np.float32)).to(self.device)
+        output = self.self.speech2embedding(audio)
```

Comment:
Like L150-152, how about giving a warning and averaging it to a single channel? 
In terms of spk recognition performance, this is normally ok. 

But I'm not sure about SE and TTS.

Reply:
Sounds good, it should be fine.